cpp/src/io/parquet/reader.cpp

```diff
@@ -40,4 +40,19 @@ table_with_metadata reader::read(parquet_reader_options const& options)
                      options.get_row_groups());
 }
 
+chunked_reader::chunked_reader(std::size_t chunk_read_limit,
+                               std::vector<std::unique_ptr<datasource>>&& sources,
+                               parquet_reader_options const& options,
+                               rmm::cuda_stream_view stream,
+                               rmm::mr::device_memory_resource* mr)
+{
+  _impl = std::make_unique<impl>(chunk_read_limit, std::move(sources), options, stream, mr);
+}
```

Comment:
```suggestion
                               rmm::mr::device_memory_resource* mr):
                                 _impl{std::make_unique<impl>(chunk_read_limit, std::move(sources), options, stream, mr)}{}
```

Reply:
`_impl` is in the based class thus we can't initialize it in the constructor of this derived class.

Reply:
IMO we would be better off without any inheritance here, but maybe I'm missing some important benefit.